### ISSUE TITLE
Same link mutiple times on page gets different link hints

### ISSUE THREAD

**Issue by flub** [philc/vimium#2820]
Firefox version 52.4.0
Mozilla/5.0 (X11; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0
Vimium 1.61.1

On a page where the same link shows up multiple times, e.g. https://docs.python.org/3/reference/datamodel.html#object.__hash__ has plenty of identical links to __eq__, each links still gets a different link hint.  It would be much nicer to give them all the same link hint so that it would be obvious that the various links all go to the same page.  It also keeps the number of required link hints down, reducing the needs for longer character sequences.

**Comment by mrmr1993** [comment]
I don't think this is as clear-cut as it sounds. Pages can use the same link URL for elements with sightly (or very) different javascript behaviour to support their users who don't have javascript enabled. Without something like #1859, we generally can't distinguish a plain link from one of these scripted links.

Along these lines, I also worry that it's confusing if users have selected a link hint intending to click one element and then we indicate that we've clicked a different one, even if the results are the same.

Still, this would be a nice option, especially for link hints that only deal with links' URLs (open in incognito/ copy URL, IIRC).

**Comment by lydell** [comment]
VimFx did this with great success.

**Comment by flub** [comment]
Yeah, it was missing this experience coming from VimFX that made me file this :)

**Comment by nerton** [comment]
>  I also worry that it's confusing if users have selected a link hint intending to click one element and then we indicate that we've clicked a different one, even if the results are the same.

There should be an option for this in advanced options which advanced users can turn on.

Advanced users would appreciate having the same links labeled with the same hint and therefore having more unique single letter hints for other links.
